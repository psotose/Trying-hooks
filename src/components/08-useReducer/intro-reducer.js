const initialState = [{
  id: 1,
  todo: 'Comprar pan',
  done: false
}];

const myFirstTodoReducer = (state = initialState, action) => {

  if ( action?.type === 'add' ){
    return [ ...state, action.payload ];
  }
  return state;
}

let todos = myFirstTodoReducer();

const newTodo = [{
  id: 2,
  todo: 'Comprar leche',
  done: false
}];

const addTodoAction = {
  type: 'add',
  payload: newTodo,
}

todos = myFirstTodoReducer( todos, addTodoAction);

console.log(todos);