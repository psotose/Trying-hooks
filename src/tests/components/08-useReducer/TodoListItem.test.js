import React from 'react'
import { shallow } from 'enzyme';
import { TodoListItem } from '../../../components/08-useReducer/todoListItem';
import { demoTodos } from '../../fixtures/demoTodos';

describe('Testing <TodoListItem />', () => {
  
  const handleDelete = jest.fn();
  const handleToggle = jest.fn();

  const wrapper = shallow( 
    <TodoListItem
      todo={ demoTodos[0] }
      index={ 0 }
      handleDelete={ handleDelete }
      handleToggle={ handleToggle }
    /> 
  );

  test('should render correctly', () => {
    expect( wrapper ).toMatchSnapshot();
  });

  test('should call handleDelete function', () => {

    wrapper.find('button').simulate('click');
    expect(handleDelete).toHaveBeenCalledWith( demoTodos[0].id );
  });

  test('should call handleToggle function', () => {
    wrapper.find('p').simulate('click');
    expect(handleToggle).toHaveBeenCalledWith( demoTodos[0].id );
  });
  
  test('should show the correct text', () => {
    const p = wrapper.find('p');
    expect( p.text().trim() ).toBe(`1. ${ demoTodos[0].desc }`)
  });

  test('should have complete class if TODO.done = true', () => {
    
    const todo = demoTodos[0];
    todo.done = true;

    const wrapper = shallow( 
      <TodoListItem
        todo={ todo }
      /> 
    );

    expect(wrapper.find('p').hasClass('complete')).toBe(true);
  });

})