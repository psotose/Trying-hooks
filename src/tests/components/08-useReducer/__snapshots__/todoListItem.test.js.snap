// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <TodoListItem /> should render correctly 1`] = `
<li
  className="list-group-item"
  key="1"
>
  <p
    className="undefined"
    onClick={[Function]}
  >
    1
    . 
    Learn React
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Borrar
  </button>
</li>
`;
